tests/subsys/bluetooth/mesh/metadata_extraction/src/main.c

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+/** @file
+ *  @brief Bluetooth Mesh DFU Target role sample
+ */
+#include <zephyr/kernel.h>
+#include <zephyr/bluetooth/bluetooth.h>
+#include <zephyr/bluetooth/mesh.h>
+#include <bluetooth/mesh/models.h>
+
+
+static struct bt_mesh_health_srv health_srv;
+
+BT_MESH_HEALTH_PUB_DEFINE(health_pub, 0);
+
+static struct bt_mesh_model models[] = {
+	BT_MESH_MODEL_CFG_SRV,
+	BT_MESH_MODEL_HEALTH_SRV(&health_srv, &health_pub),
+};
+
+static struct bt_mesh_elem elements[] = {
+	BT_MESH_ELEM(1, models, BT_MESH_MODEL_NONE),
+};
+
+#ifndef CONFIG_COMP_DATA_LAYOUT_SINGLE
+static struct bt_mesh_onoff_cli onoff_cli;
+static struct bt_mesh_lvl_cli lvl_cli;
+
+static struct bt_mesh_model extra_models[] = {
+	BT_MESH_MODEL_ONOFF_CLI(&onoff_cli),
+	BT_MESH_MODEL_LVL_CLI(&lvl_cli),
+};
+
+static struct bt_mesh_elem elements_alt[] = {
+	BT_MESH_ELEM(1, models, BT_MESH_MODEL_NONE),
+	BT_MESH_ELEM(2, extra_models, BT_MESH_MODEL_NONE),
+};
+#endif
+
+#ifdef CONFIG_COMP_DATA_LAYOUT_ARRAY
+const struct bt_mesh_comp comp[2] = {
+	{
+		.cid = 1,
+		.pid = 2,
+		.vid = 3,
+		.elem = elements,
+		.elem_count = ARRAY_SIZE(elements),
+	},
+	{
+		.cid = 4,
+		.pid = 5,
+		.vid = 6,
+		.elem = elements_alt,
+		.elem_count = ARRAY_SIZE(elements_alt),
+	},
+};
+#else /* defined(COMP_DATA_LAYOUT_SINGLE) || defined(COMP_DATA_LAYOUT_MULTIPLE) */
+static const struct bt_mesh_comp comp = {
+	.cid = 1,
+	.pid = 2,
+	.vid = 3,
+	.elem = elements,
+	.elem_count = ARRAY_SIZE(elements),
+};
+#ifdef CONFIG_COMP_DATA_LAYOUT_MULTIPLE
+static const struct bt_mesh_comp comp_alt = {
+	.cid = 4,
+	.pid = 5,
+	.vid = 6,
+	.elem = elements_alt,
+	.elem_count = ARRAY_SIZE(elements_alt),
+};
+#endif /* CONFIG_COMP_DATA_LAYOUT_MULTIPLE */
+#endif /* defined(COMP_DATA_LAYOUT_SINGLE) || defined(COMP_DATA_LAYOUT_MULTIPLE) */
+
+static const uint8_t dev_uuid[16];
+
+static const struct bt_mesh_prov prov = {
+	.uuid = dev_uuid,
+};
+
+volatile bool use_alt;
```

Comment:
Why is this variable volatile and not static?

Reply:
If it's static it will be assumed (correctly) by the compiler to always be 0/false and the alternative composition data will be optimized out of the build, and not picked up by the script. It's a trick to make the compiler think it can't know which composition data to use until runtime. (LTO might break this trick, I haven't checked, but let's leave that until we potentially add LTO test to this test)

Reply:
Could you add it here as a comment?